src/web/app/pages-session/session-result-page/session-result-page.component.html

```diff
@@ -43,21 +46,39 @@ <h1>
 
 <tm-loading-retry [shouldShowRetry]="hasFeedbackSessionResultsLoadingFailed" [message]="'Failed to load results'" (retryEvent)="retryLoadingFeedbackSessionResults()">
   <div *tmIsLoading="isFeedbackSessionResultsLoading">
+    <div *ngIf="isFeedbackSessionResultsEmpty" class="mt-4">
+      <div class="alert alert-primary d-flex align-items-center justify-content-center" style="background-color: #D1ECF1;" role="alert">
+        Nothing to show. The responses are either not configured to be visible or there are no responses.
+      </div>
+    </div>
     <div class="card bg-light mt-4" *ngFor="let question of questions">
       <div id="question-{{ question.feedbackQuestion.questionNumber }}-responses" class="card-body">
         <tm-question-text-with-info [questionNumber]="question.feedbackQuestion.questionNumber" [questionDetails]="question.feedbackQuestion.questionDetails" class="question-text"></tm-question-text-with-info>
-        <div id="other-responses" *ngIf="canStudentSeeResponses(question)">
+        <div id="other-responses" *ngIf="canStudentSeeResponses(question), else nonViewableResponse">
           <tm-single-statistics [question]="question.feedbackQuestion.questionDetails"
-                                [responses]="question.allResponses"
-                                [statistics]="question.questionStatistics"
-                                [isStudent]="true"></tm-single-statistics>
-          <tm-student-view-responses *ngIf="question.responsesToSelf.length" [responses]="question.responsesToSelf" [feedbackQuestion]="question.feedbackQuestion" [timezone]="session.timeZone" [statistics]="question.questionStatistics"></tm-student-view-responses>
-          <div *ngFor="let responsesForOtherRecipient of question.otherResponses">
-            <tm-student-view-responses *ngIf="responsesForOtherRecipient.length" [responses]="responsesForOtherRecipient" [feedbackQuestion]="question.feedbackQuestion" [timezone]="session.timeZone" [statistics]="question.questionStatistics"></tm-student-view-responses>
+                              [responses]="question.allResponses"
+                              [statistics]="question.questionStatistics"
+                              [isStudent]="true"></tm-single-statistics>
+          <div *ngIf="question.questionStatistics, else dontShowAdditionalMsg">
```

Comment:
This `question.questionStatistics` check looks a little hardcoded and I am not sure if this is even correct. We should expose this field somehow in the API return:
```
    public boolean isIndividualResponsesShownToStudents() {
        return true;
```

However, I am not sure if this addition is easy. We can use a constant map otherwise too for now and KIV this.

Reply:
I'm also not sure how I can go about doing this. May I ask what is a constant map that you said? I can maybe try to implement that.

Reply:
just have a class variable in `session-result-page-component.ts` with this comment:
```
//isIndividualResponsesShownToStudents() is false for these questions
RESPONSE_HIDDEN_QUESTIONS = [
      "CONTRIB"
]
```

Then check if question.feedbackQuestion.questionType is in the array above